### PR TITLE
fix: prevent_large_files flag for energy_results

### PR DESCRIPTION
**Pull request by slb-halfspace**
# Description

Currently the prevent_large_files flag that creates a chunk_index is only for time_series functions. 
This PR changes that, and adds a unit_test checking it's usage on energy_results.

## Pull-request quality

- [ ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

**Review by JesperGomes — APPROVED** [review]

File: source/databricks/settlement_report/settlement_report_job/domain/report_name_factory.py [hunk 1]
```diff
@@ -38,16 +38,19 @@ def create(
...
-            return self._create_result_filename(grid_area_code, energy_supplier_id)
+            return self._create_energy_result_filename(
+                grid_area_code, energy_supplier_id, chunk_index
+            )
         else:
             raise NotImplementedError(
                 f"Report data type {self.report_data_type} is not supported."
             )
 
-    def _create_result_filename(
```
Comment (JesperGomes) [inline source/databricks/settlement_report/settlement_report_job/domain/report_name_factory.py:47]: It is shared with wholesale_results?

Comment (slb-halfspace) [reply source/databricks/settlement_report/settlement_report_job/domain/report_name_factory.py:47]: It is currently only used for energy_results, so figured it was a better name.

**Review by JesperGomes — DISMISSED** [review]

File: source/databricks/settlement_report/settlement_report_job/domain/report_name_factory.py (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/settlement_report/settlement_report_job/domain/report_name_factory.py:47]: see line 39

File: source/databricks/settlement_report/settlement_report_job/utils.py [hunk 2]
```diff
@@ -284,3 +285,32 @@ def merge_files(
...
+) -> bool:
+    return (
+        _check_if_only_one_grid_area_is_selected(
+            calculation_id_by_grid_area, grid_area_codes
+        )
+        or split_report_by_grid_area
+    )
+
+
+def _check_if_only_one_grid_area_is_selected(
```
Comment (BjarkeMeier) [inline source/databricks/settlement_report/settlement_report_job/utils.py:303]: What this function does is not intuitive based on it's name. Does it e.g. throw if more grid area are selected?
```suggestion
def _is_exactly_one_grid_area_selected(
```

**Review by JesperGomes — APPROVED** [review]

File: source/databricks/settlement_report/settlement_report_job/utils.py (hunk 2 above)
Comment (JesperGomes) [reply source/databricks/settlement_report/settlement_report_job/utils.py]: I have renamed it to your suggestion, Bjarke

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**